Intersect.Server/Entities/Entity.cs

```diff
@@ -460,110 +460,124 @@ public virtual int CanMove(Direction moveDir)
                 var tileAttribute = mapController.Attributes[tileX, tileY];
                 if (tileAttribute != null)
                 {
-                    if (tileAttribute.Type == MapAttribute.Blocked || (tileAttribute.Type == MapAttribute.Animation && ((MapAnimationAttribute)tileAttribute).IsBlock))
+                    switch (tileAttribute.Type)
                     {
-                        return -2;
-                    }
+                        case MapAttribute.Blocked:
+                        case MapAttribute.Animation when 
+                            ((MapAnimationAttribute)tileAttribute).IsBlock:
+                        case MapAttribute.NpcAvoid when
+                            (this is Npc || (this is EventPageInstance evtPage && !evtPage.MyPage.IgnoreNpcAvoids)):
+                            blockerType = MovementBlockerType.MapAttribute;
+                            entityType = EntityType.GlobalEntity;
+                            return false;
 
-                    if (tileAttribute.Type == MapAttribute.NpcAvoid && (this is Npc || (this is EventPageInstance evtPage && !evtPage.MyPage.IgnoreNpcAvoids)))
-                    {
-                        return -2;
-                    }
+                        case MapAttribute.ZDimension when
+                            ((MapZDimensionAttribute)tileAttribute).BlockedLevel > 0 &&
+                            ((MapZDimensionAttribute)tileAttribute).BlockedLevel - 1 == Z:
+                            blockerType = MovementBlockerType.ZDimension;
+                            entityType = EntityType.GlobalEntity;
+                            return false;
 
-                    if (tileAttribute.Type == MapAttribute.ZDimension &&
-                        ((MapZDimensionAttribute)tileAttribute).BlockedLevel > 0 &&
-                        ((MapZDimensionAttribute)tileAttribute).BlockedLevel - 1 == Z)
-                    {
-                        return -3;
-                    }
+                        case MapAttribute.Slide:
+                            switch (((MapSlideAttribute)tileAttribute).Direction)
```

Comment:
After checking this up and seeing the values we are getting from this, and the fact that all directions are supported and limited by the core when configured, i've realized that we can reduce this down to simply:

```csharp

case MapAttribute.Slide:
                    blockerType = MovementBlockerType.Slide;
                    entityType = EntityType.GlobalEntity;
                    return false;
```

tested with slide map attributes to be working as expected

**Also, a self reminder:** we should implement diagonal slide attributes since we can surely support them now like we do with knockbacks and dash.

![imagen](https://user-images.githubusercontent.com/17498701/228302391-f4501892-3325-4eec-b7b9-b5289f9eedc4.png)